### PR TITLE
[web][speech] Improve expo speech for web.

### PR DESCRIPTION
**Pull request by Federkun**
# Why

As `getVoices` is populated asynchronous, on web, it may return an empty array. See https://stackoverflow.com/questions/49506716/speechsynthesis-getvoices-returns-empty-array-on-windows

On native it's possible to provide a `voice`'s identifier as part of the options, to select a different voice. This PR allow web to do the same thing.

# How

-

# Test Plan

I applied this as a patch and tested on chrome (only chrome).

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-speech/src/Speech/ExponentSpeech.web.ts [hunk 1]
```diff
@@ -32,9 +49,19 @@ export default {
...
+    if (typeof options.voice === 'string') {
+      const voices = await getVoices();
+      message.voice =
+        voices[
+          Math.max(
+            0,
+            voices.findIndex((voice) => voice.voiceURI === options.voice)
+          )
+        ];
+    }
```
Comment (Federkun) [inline packages/expo-speech/src/Speech/ExponentSpeech.web.ts:64]: If the voice provided in the options don't match any available voice we fallback to the first one in the list.
Not doing any clever stuff with `voice.default` here, as `_voiceIndex` does the same

File: packages/expo-speech/src/Speech/ExponentSpeech.web.ts [hunk 2]
```diff
@@ -64,8 +91,8 @@ export default {
 
     return message;
   },
-  getVoices(): WebVoice[] {
-    const voices = window.speechSynthesis.getVoices();
+  async getVoices(): Promise<WebVoice[]> {
+    const voices = await getVoices();
```
Comment (Federkun) [inline packages/expo-speech/src/Speech/ExponentSpeech.web.ts:95]: This is fine to be async. It's used in an async function already:

```
export async function getAvailableVoicesAsync(): Promise<Voice[]> {
  if (!ExponentSpeech.getVoices) {
    throw new UnavailabilityError('Speech', 'getVoices');
  }
  return ExponentSpeech.getVoices(); // <--
}
```

File: packages/expo-speech/src/Speech/ExponentSpeech.web.ts [hunk 3]
```diff
@@ -5,6 +5,23 @@ import { SpeechOptions, WebVoice, VoiceQuality } from './Speech.types';
...
+  return new Promise<SpeechSynthesisVoice[]>((resolve) => {
+    const voices = window.speechSynthesis.getVoices();
+
+    if (voices.length > 0) {
+      resolve(voices);
+      return;
+    }
+
+    // when a page loads it takes some amount of time to populate the voices list
+    window.speechSynthesis.onvoiceschanged = function () {
```
Comment (bbarthec) [inline packages/expo-speech/src/Speech/ExponentSpeech.web.ts:18]: Let's add this StackOverflow question for reference here, shall we? 🤔 
```suggestion
    // when a page loads it takes some amount of time to populate the voices list
    // see https://stackoverflow.com/a/52005323/4337317
    window.speechSynthesis.onvoiceschanged = function () {
```
Applying this suggestion would require rebuilding the `build` directory 😞

**Review by bbarthec — CHANGES_REQUESTED** [review]
@Federkun, great job on this one! 👏 🥳 
It looks rock solid 🪨 
Let's add this single SO reference to the comment as well and we're ready to merge 😉

**Comment by Federkun** [comment]
Brilliant, added, thanks you so much!

**Review by bbarthec — APPROVED** [review]

**Pull request merged**